lib/jcsda-emc/spack-stack/stack/meta_modules.py

```diff
@@ -116,6 +116,10 @@ def get_matched_dict(root_dir, candidate_list, sub_candidate_list=None):
                     sub_matched_dict = get_matched_dict(
                         os.path.join(root_dir, matched_name, matched_version),
                         sub_candidate_list)
+            # Partial matches: name-version/ or name-version-hash/ or ...
+            elif '@' in candidate and xdir.startswith(candidate.replace('@','-')):
+                matched_name = candidate.split('@')[0]
```

Comment:
FYI: I think you can do this:
```
    (matched_name, matched_version, leftovers) = cadidate.split('@')
```
to avoid calling split multiple times. Probably doesn't really matter in this case.